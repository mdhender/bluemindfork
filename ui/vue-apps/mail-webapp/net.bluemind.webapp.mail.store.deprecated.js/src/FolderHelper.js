/**
 * @param {Array} mailboxes like:
 *    [{ mailboxUid: "mbuid",
 *        type:"mailshare",
 *        root:"Groot",
 *        folders: [{value: { fullName: "m-folder1-blue" }, parentUid: mailshareMailbox}] }]
 */
export function applyFilterThenSliceAndTransform(mailboxes, filterFunction, maxFolders) {
    return mailboxes.reduce((matches, mailbox) => {
        return matches.concat(_filter(mailbox, filterFunction, maxFolders - matches.length));
    }, []);
}

function _filter(mailbox, filter, max) {
    const matches = [];
    for (let i = 0; i < mailbox.folders.length && matches.length < max; i++) {
        const folder = mailbox.folders[i];
        const root = folder.parent !== null ? mailbox.root : "";
        if (filter(folder, Object.assign({}, mailbox, { root }))) {
            const folderItem = toFolderItem(
                folder,
                mailbox.type === "mailshare",
                ((root && root + "/") || "") + folder.path
            );
            matches.push(folderItem);
        }
    }
    return matches;
}

export function toFolderItem(folder, isShared = false, path = undefined) {
    return { folder, isShared, path };
}

export const FolderHelper = { applyFilterThenSliceAndTransform, toFolderItem };
export default FolderHelper;
