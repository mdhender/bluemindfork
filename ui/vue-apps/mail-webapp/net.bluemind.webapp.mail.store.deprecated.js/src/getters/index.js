export * from "./areAllMessagesSelected";
export * from "./areAllSelectedMessagesFlagged";
export * from "./areAllSelectedMessagesRead";
export * from "./areAllSelectedMessagesUnflagged";
export * from "./areAllSelectedMessagesUnread";
export * from "./areMessagesFiltered";
export * from "./areRemoteImagesUnblocked";
export * from "./currentFolder";
export * from "./currentMailbox";
export * from "./isMessageSelected";
export * from "./isReadOnlyFolder";
export * from "./lastRecipients";
export * from "./mailshares";
export * from "./messageListMode";
export * from "./my";
export * from "./nextMessageKey";
export * from "./status";
export * from "./unreadCount";
