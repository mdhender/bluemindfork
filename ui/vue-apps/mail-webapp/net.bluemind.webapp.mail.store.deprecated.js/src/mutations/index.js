export * from "./addAllToSelectedMessages";
export * from "./addSelectedMessageKey";
export * from "./collapseFolder";
export * from "./setShowBlockedImagesAlert";
export * from "./deleteAllSelectedMessages";
export * from "./deleteSelectedMessageKey";
export * from "./expandFolder";
export * from "./setCurrentFolder";
export * from "./setMaxMessageSize";
export * from "./setMessageFilter";
export * from "./setUnreadCount";
export * from "./setUserUid";
export * from "./setStatus";
export * from "./setUserSettings";
export * from "./toggleEditFolder";
export * from "./unblockRemoteImages";
