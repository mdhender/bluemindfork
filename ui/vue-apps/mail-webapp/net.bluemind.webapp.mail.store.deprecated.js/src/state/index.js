import { STATUS } from "../constants";

export let currentFolderKey;
export let foldersData = {};
export let login;
export let maxMessageSize;
export let messageFilter;
export let messagesWithUnblockedRemoteImages = [];
export let selectedMessageKeys = [];
export let showBlockedImagesAlert = false;
export const status = STATUS.IDLE;
export let userSettings = {};
