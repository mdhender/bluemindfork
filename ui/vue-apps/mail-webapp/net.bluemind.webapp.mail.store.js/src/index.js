export { FolderAdaptor } from "./helpers/FolderAdaptor";
export { MailboxAdaptor } from "./helpers/MailboxAdaptor";
import * as folders from "./folders";
import * as folderList from "./folderList";
import * as mailboxes from "./mailboxes";

export default {
    namespaced: true,
    state: { ...folders.state, ...folderList.state, ...mailboxes.state },
    actions: { ...folders.actions, ...mailboxes.actions },
    mutations: { ...folders.mutations, ...folderList.mutations, ...mailboxes.mutations }
};

export const ADD_FOLDER = "mail/" + folders.ADD_FOLDER;
export const ADD_FOLDERS = "mail/" + folders.ADD_FOLDERS;
export const FETCH_FOLDERS = "mail/" + folders.FETCH_FOLDERS;
export const CREATE_FOLDER = "mail/" + folders.CREATE_FOLDER;
export const RENAME_FOLDER = "mail/" + folders.RENAME_FOLDER;
export const REMOVE_FOLDER = "mail/" + folders.REMOVE_FOLDER;
export const SET_UNREAD_COUNT = "mail/" + folders.SET_UNREAD_COUNT;
export const TOGGLE_FOLDER = "mail/" + folders.TOGGLE_FOLDER;

export const TOGGLE_EDIT_FOLDER = "mail/" + folderList.TOGGLE_EDIT_FOLDER;

export const ADD_MAILBOXES = "mail/" + mailboxes.ADD_FOLDER;
export const FETCH_MAILBOXES = "mail/" + mailboxes.FETCH_MAILBOXES;
