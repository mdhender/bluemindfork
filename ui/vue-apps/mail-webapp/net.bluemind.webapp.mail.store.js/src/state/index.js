export * from "./search";
export let currentFolderKey;
export let messageFilter;
export let foldersData = {};
export let login;
export let maxMessageSize;
