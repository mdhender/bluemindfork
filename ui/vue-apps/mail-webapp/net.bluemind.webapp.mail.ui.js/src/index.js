import MailActionsPanel from "./MailActionsPanel";
import MailAlertRenderer from "./MailAlertRenderer";
import MailThread from "./MailThread/MailThread";
import MailMessageNew from "./MailMessageNew";

export { default } from "./MailApp";
export { MailActionsPanel, MailAlertRenderer, MailMessageNew, MailThread };
