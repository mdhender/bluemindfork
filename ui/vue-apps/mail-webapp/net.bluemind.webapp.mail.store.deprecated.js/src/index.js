import * as actions from "./actions";
import events from "./events/events";
import * as getters from "./getters";
import * as mutations from "./mutations";
import * as mutactions from "./mutactions";
import * as state from "./state";
import { MailboxItemsStore as messages, MailboxFoldersStore as folders } from "@bluemind/backend.mail.store";
import { ContainersStore as mailboxes } from "@bluemind/core.container.store";
import MessageStore from "./MessageStore/";
import search from "./modules/search";
export * from "./SearchHelper";
export * from "./FolderHelper";

export default {
    namespaced: true,
    state: Object.assign({}, state, events.state),
    actions: Object.assign({}, actions, mutactions, events.actions),
    mutations: Object.assign({}, mutations, events.mutations),
    getters,
    modules: {
        search,
        messages,
        folders,
        mailboxes,
        currentMessage: MessageStore,
        draft: MessageStore
    }
};
