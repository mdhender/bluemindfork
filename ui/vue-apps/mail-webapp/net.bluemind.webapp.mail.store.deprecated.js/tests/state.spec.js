import * as state from "../src/state";

describe("[Mail-WebappStore][state]", () => {
    test("initial state", () => {
        expect(state).toMatchInlineSnapshot(`
            Object {
              "currentFolderKey": undefined,
              "foldersData": Object {},
              "maxMessageSize": undefined,
              "messageFilter": undefined,
              "messagesWithUnblockedRemoteImages": Array [],
              "selectedMessageKeys": Array [],
              "showBlockedImagesAlert": false,
              "status": "idle",
              "userSettings": Object {},
              "userUid": undefined,
            }
        `);
    });
});
