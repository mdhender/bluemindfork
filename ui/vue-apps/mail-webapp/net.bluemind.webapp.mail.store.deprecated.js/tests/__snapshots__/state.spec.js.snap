// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Mail-WebappStore][state] initial state 1`] = `
Object {
  "currentFolderKey": undefined,
  "foldersData": Object {},
  "maxMessageSize": undefined,
  "messageFilter": undefined,
  "messagesWithUnblockedRemoteImages": Array [],
  "selectedMessageKeys": Array [],
  "showBlockedImagesAlert": false,
  "status": "idle",
  "userSettings": Object {},
  "userUid": undefined,
}
`;
